File: Stores/Store/StoreKind.swift
// Copyright (c) 2017-2018 Coinbase Inc. See LICENSE

import Foundation

/// Represents the kind of store.
public enum StoreKind: String {
    /// Store data to Keychain.
    case keychain

    /// Stores data to UserDefaults.
    case userDefaults

    /// Store data to memory.
    case memory
}

File: android/store/src/main/java/com/coinbase/wallet/store/models/StoreKind.kt
package com.coinbase.wallet.store.models

enum class StoreKind {
    /**
     * Stores plaintext key/value entries in Android's SharedPreferences
     */
    SHARED_PREFERENCES,

    /**
     * Stores plaintext key/value entries in memory. Mainly used as a simple caching solution
     */
    MEMORY,

    /**
     * Stores encrypted value in Android's SharedPreferences using Android KeyStore
     */
    ENCRYPTED_SHARED_PREFERENCES
}